standard/classes.md

```diff
@@ -412,23 +413,32 @@ Each *type_parameter_constraints_clause* consists of the token `where`, followed
 
 The list of constraints given in a `where` clause can include any of the following components, in this order: a single primary constraint, one or more secondary constraints, and the constructor constraint, `new()`.
 
-A primary constraint can be a class type or the ***reference type constraint*** `class` or the ***value type constraint*** `struct`. A secondary constraint can be a *type_parameter* or *interface_type*.
+A primary constraint can be a class type, the ***reference type constraint*** `class`, the ***value type constraint*** `struct`, or the ***unmanaged type constraint*** `unmanaged`.
+
+A secondary constraint can be a *type_parameter* or *interface_type*.
 
 The reference type constraint specifies that a type argument used for the type parameter shall be a reference type. All class types, interface types, delegate types, array types, and type parameters known to be a reference type (as defined below) satisfy this constraint.
 
 The value type constraint specifies that a type argument used for the type parameter shall be a non-nullable value type. All non-nullable struct types, enum types, and type parameters having the value type constraint satisfy this constraint. Note that although classified as a value type, a nullable value type ([§8.3.11](types.md#8311-nullable-value-types)) does not satisfy the value type constraint. A type parameter having the value type constraint shall not also have the *constructor_constraint*, although it may be used as a type argument for another type parameter with a *constructor_constraint*.
 
 > *Note*: The `System.Nullable<T>` type specifies the non-nullable value type constraint for `T`. Thus, recursively constructed types of the forms `T??` and `Nullable<Nullable<T>>` are prohibited. *end note*
 
-Pointer types are never allowed to be type arguments and are not considered to satisfy either the reference type or value type constraints.
+The token `unmanaged` is neither a keyword nor a contextual keyword. When it is encountered, it will either:
```

Comment:
This might be inconsistent with the approach taken for `nameof` (§11.7.20) which is a contextual keyword (§6.4.4) and hence an *identifier* (§6.4.3) – this might stem from different ways the MS docs and the Standard spec contextual keywords and identifiers.

If we follow the approach for `nameof` then `unmanaged` would be added as an alternative to `contextual_keyword` and lines 426-429 could be replaced by a paragraph echoing the one for resolving `nameof` in §11.7.20, e.g. (written by copying the `nameof` para and editing it!):

> Because `unmanaged` is not a keyword, in *primary_constraint* the unmanaged constraint is always syntactically ambiguous with *class_type*. For compatibility reasons, if a name lookup (§11.7.4) of the name `unmanaged` succeeds it is treated as a `class_type`. Otherwise it is treated as the unmanaged constraint.

Reply:
Agreed - this is also similar to `var` (and `dynamic`, although I now can't see where that's actually specified).

It definitely sounds like the description in https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/lexical-structure.md#644-keywords - if there's a reason why it's *not* a contextual keyword, I'd like to hear it and we can include that in the standard.